cuappdev/old-eatery-blue-backend
# Implement new menu DB schema and menu fetch calculation

In this PR I:
- Refactored the DayOfWeekEventSchedule model to include a start date, repeat interval, and offset list to represent menus that repeat in any type of pattern (a bi-weekly menu, for example, could be represented as having a repeat interval of 14 and an offset of 0 if it started today and repeated on Fridays bi-weekly)

- Note that the offset list is represented as a CharField, but I applied a validator to it that ensures the data is always stored as a comma separated int list, so it essentially mimics an array-like field 

- Refactored the DayOfWeekSchedule DfgNode to determine which events are happening on a particular day for an eatery via its start date, repeat interval, and offset list

- Renamed all references to DayOfWeekEventSchedule and DayOfWeekSchedule to RepeatingEventSchedule and RepeatingSchedule because it makes more sense now that we are not constraining ourselves to weekly menus. 

To test:
- Run bash reset_db.sh to restructure the DB tables

- Add default values for the offsets, start date, and repeat interval when prompted by Django (note that validation checks won't apply here so make sure the offsets are a comma separated string of integers with no spaces, i.e., "1,3,5")

- Change the amount of days queried over from 7 to whatever value is necessary in views.py and query /api
